app/controllers/service_provider_controller.rb

```diff
@@ -2,11 +2,27 @@ class ServiceProviderController < ApplicationController
   protect_from_forgery with: :null_session
 
   def update
-    if FeatureManagement.use_dashboard_service_providers?
-      ServiceProviderUpdater.new.run
-      SecureHeadersWhitelister.new.run
+    authorize do
+      if FeatureManagement.use_dashboard_service_providers?
+        ServiceProviderUpdater.new.run
+        SecureHeadersWhitelister.new.run
+      end
+
+      render json: { status: 'If the feature is enabled, service providers have been updated.' }
+    end
+  end
+
+  private
+
+  def authorize
+    if authorization_token == Figaro.env.dashboard_api_token
+      yield
+    else
+      render nothing: true, status: 401
     end
+  end
 
-    render json: { status: 'If the feature is enabled, service providers have been updated.' }
+  def authorization_token
+    request.headers['X-LOGIN-DASHBOARD-TOKEN']
```

Comment:
It looks from that link like rack does not transform headers but that the expected format is underscored and prefix with `HTTP`. Am I missing something?

Totally down to use that format if it is the expected standard. 

Interestingly, @monfresh @pkarman and I had a detailed convo about header naming ideas in the issue associated with this PR. Of all the suggestions that we made based on past app development experience, none of them use this convention! (all were dasherized and none started with `http`)